### ISSUE TITLE
CultureInfo.GetCultureInfo(@"pt-PT") returns incorrect case for day name

### ISSUE THREAD

**Issue by nealef** [mono/mono#20111]
## Steps to Reproduce

1. Build and run the following program:
```
using System;
using System.Text;
using System.Globalization;
using System.IO;

public class Tests {

	static void Main(string[] args)
        {

            DateTime when = DateTime.UtcNow;
            Console.WriteLine(when.Date.ToString(@"ddd, d MMM", CultureInfo.GetCultureInfo(@"pt-PT")));
            Console.Read();
        }
}
```
### Current Behavior

Returns:
```
terça, 14 jul
```
This is the equivalent of ```Tuesday, 14 jul```

### Expected Behavior

Running with .NET on Windows 10:
```
ter, 14 jul
```
This is the equivalent of ```tuesday, 14 jul```

## On which platforms did you notice this

[ ] macOS
[X] Linux
[ ] Windows

**Version Used**:

5.16 and master

## Stacktrace

N/A

### Further Information

If I use ```pt-BR``` instead of ```pt-PT``` it returns the correct information. It has me wondering if this actually a problem with ICU in Linux rather than a mono issue, but am reporting in case someone has insight. I will investigate using the equivalent program in C.

**Comment by nealef** [comment]
The problem is not present in 4.*x*.

I wondered if the locale data may have been out of date so I attempted to remake ```culture-table-info.h``` by rebuilding and running ```tools/locale-builder``` but the data looked incomplete so I grabbed a more recent CLDR (37) and added it to the existing 30.0.2 material (as some of the locale files seem to have been deleted in the later release). I needed to add some entries to the ```known_LCIDs``` hashset (and rename ```ca_ES_valencia``` to ```ca_ES_VALENCIA```. Other than it complaining about missing locale files it crashed thusly:
```
Unhandled Exception:
System.Collections.Generic.KeyNotFoundException: The given key '4096' was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x0001e] in <5ac37efc89440a43a4a1a47801f82d40>:0
  at Mono.Tools.LocaleBuilder.Patterns.FillValues (System.Int32 lcid, Mono.Tools.LocaleBuilder.CultureInfoEntry ci) [0x000e5] in /home/neale/mono-master/tools/locale-builder/Patterns.cs:1619
  at Mono.Tools.LocaleBuilder.Driver.Import (Mono.Tools.LocaleBuilder.CultureInfoEntry data, System.String locale) [0x0048a] in /home/neale/mono-master/tools/locale-builder/Driver.cs:997
  at Mono.Tools.LocaleBuilder.Driver.Run () [0x014df] in /home/neale/mono-master/tools/locale-builder/Driver.cs:319
  at Mono.Tools.LocaleBuilder.Driver.Main (System.String[] args) [0x0000f] in /home/neale/mono-master/tools/locale-builder/Driver.cs:51
[ERROR] FATAL UNHANDLED EXCEPTION: System.Collections.Generic.KeyNotFoundException: The given key '4096' was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x0001e] in <5ac37efc89440a43a4a1a47801f82d40>:0
  at Mono.Tools.LocaleBuilder.Patterns.FillValues (System.Int32 lcid, Mono.Tools.LocaleBuilder.CultureInfoEntry ci) [0x000e5] in /home/neale/mono-master/tools/locale-builder/Patterns.cs:1619
  at Mono.Tools.LocaleBuilder.Driver.Import (Mono.Tools.LocaleBuilder.CultureInfoEntry data, System.String locale) [0x0048a] in /home/neale/mono-master/tools/locale-builder/Driver.cs:997
  at Mono.Tools.LocaleBuilder.Driver.Run () [0x014df] in /home/neale/mono-master/tools/locale-builder/Driver.cs:319
  at Mono.Tools.LocaleBuilder.Driver.Main (System.String[] args) [0x0000f] in /home/neale/mono-master/tools/locale-builder/Driver.cs:51
```
Strangely commit 23b8f10cb89b18e2cc37ea49abfa60c3166081d3 refers to version 34 of CLDR but the ```Makefile.am``` file still has 30.0.2 in the URL from where it grabs the core.zip file.

**Comment by nealef** [comment]
It appears the abbreviated day name data in the ```pt_PT.xml``` file spells out the long form of the day name rather than the 3 letter abbreviation. I manually updated the file and the output was as expected. Note, ```pt_BR``` effectively inherits from ```pt.xml``` which does have the abbreviations. 

The sample program also illustrates another difference to running on .NET as the abbreviations are suffixed by a ```.``` such that the output for ```pt_BR``` is:
```
ter., 14 jul.
```
In addition, the ```make install-culture-table``` command copies ```culture-info-tables.h``` to ```mono/metadata``` but it seems it should be copying it to ```mono/culture``` as it is used by ```locales.c``` to build the ```libmono_culture``` shared object.

**Comment by nealef** [comment]
Note, this behaviour changed when CLDR went from v24 to v30.0.2.

**Comment by nealef** [comment]
I have asked about the change in https://unicode-org.atlassian.net/browse/CLDR-13961.

**Comment by lambdageek** [comment]
> In addition, the `make install-culture-table` command copies `culture-info-tables.h` to `mono/metadata` but it seems it should be copying it to `mono/culture` as it is used by `locales.c` to build the `libmono_culture` shared object.

Good catch.  Missed that part when making the changes in https://github.com/mono/mono/pull/19912

**Comment by lambdageek** [comment]
Made a PR to update https://github.com/mono/mono/pull/20124
